### ISSUE TITLE
Select component is not scrollable in shadow dom

### ISSUE THREAD

**Issue by sandro768** [radix-ui/primitives#1980]
## Bug report

Select component is not scrollable in shadow dom due to a bug of [react-remove-scroll](https://github.com/theKashey/react-remove-scroll)

Here is a link of that bug - https://github.com/theKashey/react-remove-scroll/issues/45

### Reproducible example

[CodeSandbox Template](https://codesandbox.io/s/heuristic-dawn-dwq3mi?file=/src/App.js)

**Comment by focux** [comment]
I'm also experiencing this when using `position` popper. Did you find any workaround @ColmanHumphrey?

**Comment by ColmanHumphrey** [comment]
> I'm also experiencing this when using `position` popper. Did you find any workaround @ColmanHumphrey?

Hey @focux — I think we ended up setting a max height. It's not ideal because in theory you have to set it to be less than half the screen (for when the trigger is in the middle), and that looks bad when the trigger is up top. But sure that's life.

**Comment by joaom00** [comment]
Have you tried using `--radix-select-content-available-height`?

https://www.radix-ui.com/docs/primitives/components/select#constrain-the-content-size

**Comment by focux** [comment]
Just tried and it didn't work.

**Comment by joaom00** [comment]
It should work :thinking:. Is your select in a shadow dom too?

**Comment by kyrylolvov** [comment]
> It should work 🤔. Is your select in a shadow dom too?

Will your approach work? What is the example usage of that css variable in the component?

**Comment by kyrylolvov** [comment]
I was able to fix the issue by adding a data-scrollable to part of the dropdown content which should be scrollable and then adding this event listeners.

`	useEffect(() => {
		const handleWheel = (e: WheelEvent) => {
			if ((e.target as Element).closest('[data-scrollable]')) return;
			e.stopPropagation();
		};

		const handleTouchMove = (e: TouchEvent) => {
			if ((e.target as Element).closest('[data-scrollable]')) return;
			e.stopPropagation();
		};

		document.addEventListener('wheel', handleWheel, true);
		document.addEventListener('touchmove', handleTouchMove, true);

		return () => {
			document.removeEventListener('wheel', handleWheel, true);
			document.removeEventListener('touchmove', handleTouchMove, true);
		};
	}, []);
`

**Comment by f3sa** [comment]
> I was able to fix the issue by adding a data-scrollable to part of the dropdown content which should be scrollable and then adding this event listeners.
> 
> ```
> useEffect(() => {
>     const handleWheel = (e: WheelEvent) => {
>         if ((e.target as Element).closest('[data-scrollable]')) return;
>         e.stopPropagation();
>     };
> 
>     const handleTouchMove = (e: TouchEvent) => {
>         if ((e.target as Element).closest('[data-scrollable]')) return;
>         e.stopPropagation();
>     };
> 
>     document.addEventListener('wheel', handleWheel, true);
>     document.addEventListener('touchmove', handleTouchMove, true);
> 
>     return () => {
>         document.removeEventListener('wheel', handleWheel, true);
>         document.removeEventListener('touchmove', handleTouchMove, true);
>     };
> }, []);
> ```

Could you be kind to share your code on where you pasted the data-scrollable and where you inserted the useEffect?

**Comment by kyrylolvov** [comment]
> > I was able to fix the issue by adding a data-scrollable to part of the dropdown content which should be scrollable and then adding this event listeners.
> > ```
> > useEffect(() => {
> >     const handleWheel = (e: WheelEvent) => {
> >         if ((e.target as Element).closest('[data-scrollable]')) return;
> >         e.stopPropagation();
> >     };
> > 
> >     const handleTouchMove = (e: TouchEvent) => {
> >         if ((e.target as Element).closest('[data-scrollable]')) return;
> >         e.stopPropagation();
> >     };
> > 
> >     document.addEventListener('wheel', handleWheel, true);
> >     document.addEventListener('touchmove', handleTouchMove, true);
> > 
> >     return () => {
> >         document.removeEventListener('wheel', handleWheel, true);
> >         document.removeEventListener('touchmove', handleTouchMove, true);
> >     };
> > }, []);
> > ```
> 
> Could you be kind to share your code on where you pasted the data-scrollable and where you inserted the useEffect?

Hello! So the useEffect I put in App.tsx (I'm using React.js), basically it should run on app mount. And as for the  data-scrollable, I just passed it as prop to element inside DropdownContent which should be scrollable.

```
<div className="max-h-[210px] overflow-y-scroll" data-scrollable>
...
</div
```

**Comment by f3sa** [comment]
> > > I was able to fix the issue by adding a data-scrollable to part of the dropdown content which should be scrollable and then adding this event listeners.
> > > ```
> > > useEffect(() => {
> > >     const handleWheel = (e: WheelEvent) => {
> > >         if ((e.target as Element).closest('[data-scrollable]')) return;
> > >         e.stopPropagation();
> > >     };
> > > 
> > >     const handleTouchMove = (e: TouchEvent) => {
> > >         if ((e.target as Element).closest('[data-scrollable]')) return;
> > >         e.stopPropagation();
> > >     };
> > > 
> > >     document.addEventListener('wheel', handleWheel, true);
> > >     document.addEventListener('touchmove', handleTouchMove, true);
> > > 
> > >     return () => {
> > >         document.removeEventListener('wheel', handleWheel, true);
> > >         document.removeEventListener('touchmove', handleTouchMove, true);
> > >     };
> > > }, []);
> > > ```
> > 
> > 
> > Could you be kind to share your code on where you pasted the data-scrollable and where you inserted the useEffect?
> 
> Hello! So the useEffect I put in App.tsx (I'm using React.js), basically it should run on app mount. And as for the data-scrollable, I just passed it as prop to element inside DropdownContent which should be scrollable.
> 
> ```
> <div className="max-h-[210px] overflow-y-scroll" data-scrollable>
> ...
> </div
> ```

Thank you, it works flawlessly!

**Comment by alexgshaw** [comment]
For reference adding this class worked for me.

`max-h-[var(--radix-select-content-available-height)]`

**Comment by benoitgrelard** [comment]
Fixed by #2765
